session/txn.go

```diff
@@ -456,6 +470,67 @@ func (txn *LazyTxn) LockKeysFunc(ctx context.Context, lockCtx *kv.LockCtx, fn fu
 	return txn.Transaction.LockKeysFunc(ctx, lockCtx, lockFunc, keys...)
 }
 
+// StartAggressiveLocking wraps the inner transaction to support using aggressive locking with lazy initialization.
+func (txn *LazyTxn) StartAggressiveLocking() {
+	if txn.Valid() {
+		txn.Transaction.StartAggressiveLocking()
+	} else if txn.pending() {
+		txn.enterAggressiveLockingOnValid = true
+	} else {
+		panic("trying to start aggressive locking on a transaction in invalid state")
+	}
+}
+
+// RetryAggressiveLocking wraps the inner transaction to support using aggressive locking with lazy initialization.
+func (txn *LazyTxn) RetryAggressiveLocking(ctx context.Context) {
+	if txn.Valid() {
+		txn.Transaction.RetryAggressiveLocking(ctx)
+	} else if !txn.pending() {
+		panic("trying to retry aggressive locking on a transaction in invalid state")
+	}
+}
+
+// CancelAggressiveLocking wraps the inner transaction to support using aggressive locking with lazy initialization.
+func (txn *LazyTxn) CancelAggressiveLocking(ctx context.Context) {
+	if txn.Valid() {
+		txn.Transaction.CancelAggressiveLocking(ctx)
+	} else if txn.pending() {
+		if txn.enterAggressiveLockingOnValid {
+			txn.enterAggressiveLockingOnValid = false
+		} else {
+			panic("trying to cancel aggressive locking when it's not started")
+		}
+	} else {
+		panic("trying to cancel aggressive locking on a transaction in invalid state")
+	}
+}
+
+// DoneAggressiveLocking wraps the inner transaction to support using aggressive locking with lazy initialization.
+func (txn *LazyTxn) DoneAggressiveLocking(ctx context.Context) {
+	if txn.Valid() {
+		txn.Transaction.DoneAggressiveLocking(ctx)
+	} else if txn.pending() {
+		if txn.enterAggressiveLockingOnValid {
+			txn.enterAggressiveLockingOnValid = false
+		} else {
+			panic("trying to finish aggressive locking when it's not started")
+		}
+	} else {
+		panic("trying to cancel aggressive locking on a transaction in invalid state")
+	}
+}
+
+// IsInAggressiveLockingMode wraps the inner transaction to support using aggressive locking with lazy initialization.
+func (txn *LazyTxn) IsInAggressiveLockingMode() bool {
+	if txn.Valid() {
+		return txn.Transaction.IsInAggressiveLockingMode()
+	}
```

Comment:
```suggestion
	if txn.Valid() {
	} else if if txn.pending() {
	}
```
looks better we won't miss anything.

Reply:
🤔Actually I think it doesn't matter much since there are early-returns in every branches.